### ISSUE TITLE
License issue

### ISSUE THREAD

**Issue by a3kov** [secretkeysio/jelly-actix-web-starter#9]
First, thanks for your work, and please don't take it as a complaint
This template depends on background-jobs, which is not a free software (it's under restrictive copyleft license called Cooperative Software License). I think it brings 2 points
1. It would be nice for the users to know about this fact so that they don't fall into a trap
2. Am not a legal expert but It might not even compatible with your ""do-whatever-you-want" license.

**Comment by ryanmcgrath** [comment]
Ah, you have a very good point here. Should at the very least be a note...

I think it's probably fine with the license on the repo (given that it's just a code dump) but I'll earmark time this evening to look into it more.

**Comment by ryanmcgrath** [comment]
Eh, yeah, I think you're correct and it's worth changing out this part. I'll look at doing so tonight.

Sigh, lol

**Comment by ryanmcgrath** [comment]
Note for myself: https://crates.io/crates/sqlxmq seems like it might be a decent fit, given that this repo already uses `sqlx` and `postgres` anyway. The DB-backed bit is slightly less ideal for some higher throughput use-cases but it's perhaps overkill to optimize for that here.

**Comment by mikeumus** [comment]
Another hidden gem! 💎

**Comment by ryanmcgrath** [comment]
Spent an hour or two on this tonight, will probably have something in the next day or so. It might require a pull request to the sqlxmq crate, so in the meantime it might wind up pointing to a local fork.

**Comment by ryanmcgrath** [comment]
Will leave this open until the pull request for `sqlxmq` support can be merged; `actix-web 4.0`'s betas are currently in a weird state and need some fixing, and they're required due to a combination of `sqlx 0.5.2` and Tokio `1.0`.

Users who want to take a shot at making the refresh branch work are more than welcome to. In the meantime I'll put a note in the README about this issue and link to this comment.

**Comment by a3kov** [comment]
I have a local version of this template ported to beta5. Could make a PR but probably it's better to wait for the final release. Beta6 appears to have some breaking changes, it's not clear if it was intended or not, cause multiple people have trouble with it

**Comment by ryanmcgrath** [comment]
Haha, yeah... a different app I use is a Diesel one, I can see the appeal. My compile times are just much leaner with sqlx so I opt for it these days.

Probably just waiting for 4.0 to drop is easiest, but I'd be curious if your selected actix-web beta versions would mesh with the refresh branch.

Whatever the case, super appreciate you noting this issue.

**Comment by a3kov** [comment]
Well Diesel is ORM with ability to run raw sql...sqlx is the latter, so it's not even in the same league. You absolutely need some form of query builder for dynamic queries, especially with composable software where different pieces plug in their own sql parts (filters, etc). Plus the whole async for relational databases looks like a footgun - thread pool seems to be just fine

**Comment by ryanmcgrath** [comment]
I agree - if I was building a true Django clone, I'd probably reach for Diesel (i.e, I don't think you could build an automatic-admin with raw SQL very easily). I find raw SQL to be more than adequate for whatever I wind up building these days, though... never really feels limiting, and I do find Diesel's debugging experience to be frustrating at points.

To each their own, though. Diesel's a very valid approach.

**Comment by a3kov** [comment]
@mikeumus yes that one is problematic

**Comment by mikeumus** [comment]
@a3kov thanks for the clarification. I'll just stick with sqlx for now then. 
Looking forward to it! Thank you 🙇🏻‍♂️

**Comment by soyayaos** [comment]
```background-jobs``` [0.11.0](https://crates.io/crates/background-jobs/0.11.0) licensed under GPL.

**Comment by ssokolow** [comment]
As the GNU project [touches on](https://www.gnu.org/licenses/license-list.html#informal) Your "do whatever you want" license (formally known as an "informal license") assumes United States jurisprudence and it's *not* something I'd advise or trust my codebase to depend on, since I'd be risking it serving as legal poison to *other* people who want to use *my* derived codebases.

Give this English-language legal analysis of the CC0 for suitability in Germany a read for an example of why it's not enough:

https://rd-alliance.org/sites/default/files/cc0-analysis-kreuzer.pdf

(For example, German law doesn't have a concept of releasing things into the public domain prematurely. That's why the CC0 has an "if that's not possible, here's an explicit license that gives you equivalent permissions" fallback. Likewise, German (and, more broadly, law in various EU countries) has rights that protect individuals from exploitation which can't be waived by licenses granted before they were written in to law, so the CC0 has a legally binding promise that you and your heirs won't sue over rights that are retroactively granted to you in the future.)

...and, as demonstrated by some of the things linked from [What is wrong with the Unlicense?](https://softwareengineering.stackexchange.com/questions/147111/what-is-wrong-with-the-unlicense) over on StackExchange, like [this one](https://web.archive.org/web/20170301020915/https://lists.opensource.org/pipermail/license-review/2012-January/001386.html) from the `license-review` mailing list in 2012, non-lawyers shouldn't write legal terms.

**Comment by ryanmcgrath** [comment]
> non-lawyers shouldn't write legal terms.

I'll ignore the needless dig since it's not the point here. :)

Anyway: I do acknowledge the licensing is def still an issue and this fell by the wayside, and should have been fixed.

Honestly, the issue has sat for more than a year and I don't even use Actix anymore - could (should) likely just kill the repo, which might make more sense here. If I don't get the energy or drive to finish the branch in the next day or two I'll probably just put it to bed.

**Comment by ssokolow** [comment]
It's not a dig. It's a statement of fact, backed up by the analysis of the Unlicense that I linked to which shows how problematic incredibly subtle distinctions in language can be.

"Non-lawyers shouldn't write legal terms" in the same way that people who aren't licensed to practice medicine shouldn't perform brain surgery.

**Comment by ryanmcgrath** [comment]
Sigh.

> how problematic incredibly subtle distinctions in language can be

This line is notable given that my response regarding "a dig" is actually down to how you've engaged - and in particular the way you've written and commented here. I would suggest - and encourage - you to go take a class on how to write in a way that communicates with people beyond blunt statements and a tone that weirdly tries to assume a high ground.

A simple `"hey, this has sat for x time - can we get an update or is there a course of action here?"` would've been more than sufficient and less eyeroll inducing to read. There's no need to make a remark regarding who's written what "license" when the "license" in this repo is a two-line comment representing a generic code dump. This (still valid) issue was generally a respectful tone all around and your overall tone stands out - and this is a larger problem with open source and I would invite you to contribute to fixing the issue by setting a better example. We all have to live with this.

Anyway, as I already noted I'll mull on this for the next day or two and see if I want to bother fixing it or just kill the repo entirely. You can feel free to ping if I've forgotten it again.

**Comment by Jieiku** [comment]
@ryanmcgrath If you do close the repo I would totally understand since you are no longer using actix, it would kinda make me sad a little though because I definitely learned a bit while working with this code.

I am curious if you are currently using Axum now instead of Actix, it certainly got my interest, but I have not tried working with it yet.

**Comment by ssokolow** [comment]
> A simple `"hey, this has sat for x time - can we get an update or is there a course of action here?"` would've been more than sufficient and less eyeroll inducing to read. There's no need to make a remark regarding who's written what "license" when the "license" in this repo is a two-line comment representing a generic code dump. This (still valid) issue was generally a respectful tone all around and your overall tone stands out - and this is a larger problem with open source and I would invite you to contribute to fixing the issue by setting a better example. We all have to live with this.

Would it have helped if I'd opened a new issue instead? There was no "hey, this has sat for x time" component intended and I usually would but, this time, I was tired and in a hurry to clear out a bunch of stale Reddit tabs I'd left as "to triage" and didn't think, so I just replied here because someone else had mentioned it already.

**Comment by ryanmcgrath** [comment]
@ssokolow it likely would've resulted in the same exchange, but it's not worth dwelling on now. I hear you on being tired, happens to the best of us.

@Jieiku Heh, then if nothing else I'm glad you got something out of it! Yeah, using Axum these days - while `actix` uses `tokio` under the hood now, `axum` simply feels like less moving parts. I wouldn't consider it a ringing endorsement of one over the other though and moreso a convenience feeling. Thank you again for the PRs you've submitted over the course of this repo.

I think what I'll do is make the repo private later today so that any further license issues are frozen for now - i.e, it won't result in people spreading or drawing from it until I've had time to touch it up again. I might also just frankly redo this repo in Axum if I have time in December, so we'll see - that may just supersede everything.
